src/components/Widgets/WidgetEvents.tsx

```diff
@@ -224,21 +222,31 @@ export function WidgetEvents() {
     };
 
     const onAvailableRoutes = async (availableRoutes: Route[]) => {
-      // Compare current availableRoutes with the previous one
-      if (
-        !previousRoutesRef.current ||
-        previousRoutesRef.current.numRoutes !== availableRoutes.length ||
-        previousRoutesRef.current.fromChain !== sourceChainToken.chainId ||
-        previousRoutesRef.current.fromToken !== sourceChainToken.tokenAddress ||
-        previousRoutesRef.current.toChain !== destinationChainToken.chainId ||
-        previousRoutesRef.current.toToken !== destinationChainToken.tokenAddress
-      ) {
-        // Update the previous routes ref to the current available routes
-        previousRoutesRef.current.numRoutes = availableRoutes.length;
-        previousRoutesRef.current.fromChain = sourceChainToken.chainId;
-        previousRoutesRef.current.fromToken = sourceChainToken.tokenAddress;
-        previousRoutesRef.current.toChain = destinationChainToken.chainId;
-        previousRoutesRef.current.toToken = destinationChainToken.tokenAddress;
+      // previous available routes
+      const prevRefObj = previousRoutesRef.current;
+      delete prevRefObj?.[TrackingEventParameter.TxHash];
+      // current available routes
+      const newObj = {
+        totalRoutes: availableRoutes.length,
+        fromToken: sourceChainToken.tokenAddress,
+        fromChain: sourceChainToken.chainId,
+        toToken: destinationChainToken.tokenAddress,
+        toChain: destinationChainToken.chainId,
+      };
+
+      // compare current availableRoutes with the previous one
+      const isSameObject = shallowEqualObjects(prevRefObj, newObj);
+      if (!isSameObject && previousRoutesRef.current) {
+        previousRoutesRef.current[TrackingEventParameter.AvailableRoutesCount] =
```

Comment:
not needed to do this, just do previousRoutesRef.current = newObj as you've just created an object matching this interface..